docs/programming_guide/controllers/model_controller.rst

```diff
@@ -0,0 +1,247 @@
+
+.. note::
+
+    The :ref:`fl_model` object can be any type of data.
+    For example, in the "train" and "validate" tasks we send the model parameters along with the task so the target clients can train and validate the model.
+    However in many other tasks that do not involve sending the model (e.g. "submit_model"), the :ref:`fl_model` can contain any type of data (e.g. metadata, metrics etc.) or may be not be needed at all.
+
+
+send_model_and_wait
+-------------------
+:func:`send_model_and_wait<nvflare.app_common.workflows.model_controller.ModelController.send_model_and_wait>` is the core communication function which enables users to send tasks to targets, and wait for responses.
+
+The ``data`` is an :ref:`fl_model` object, and the ``task_name`` is the task for the target executors to execute (Client API executors by default support "train", "validate", and "submit_model", however executors can be written for any arbitrary task name).
+
+``targets`` can be chosen from client names obtained with ``sample_clients()``.
+
+Returns the :ref:`fl_model` responses from the target clients once the task is completed (``min_responses`` have been received, or ``timeout`` time has passed).
+
+send_model
+----------
+:func:`send_model<nvflare.app_common.workflows.model_controller.ModelController.send_model>` is the non-blocking version of 
+:func:`send_model_and_wait<nvflare.app_common.workflows.model_controller.ModelController.send_model_and_wait>` with a user-defined callback when receiving responses.
+
+A callback with the signature ``Callable[[FLModel], None]`` can be passed in, which will be called when a response is received from each target.
+
+The task is standing until either ``min_responses`` have been received, or ``timeout`` time has passed.
+Since this call is asynchronous, the Controller :func:`get_num_standing_tasks<nvflare.apis.impl.controller.Controller.get_num_standing_tasks>` method can be used to get the number of standing tasks for synchronization purposes.
+
+
+Saving & Loading
+================
+
+persistor
+---------
+The :func:`save_model<nvflare.app_common.workflows.model_controller.ModelController.save_model>` and :func:`load_model<nvflare.app_common.workflows.model_controller.ModelController.load_model>`
+functions utilize the configured persistor set in the ModelController ``persistor_id: str = "persistor"`` argument.
+
+For the JobAPI, the persistor with ``id = "persistor"`` will automatically be configured based on the type of the model sent to the server.
+
+.. code-block:: python
+
+  job.to(Net(), "server")
+
+The persistor can also be configured in ``config_fed_server.json`` in the components section.
+
+custom save & load
+------------------
+Users can also choose to instead create their own save and load functions rather than use a persistor.
+
+For example we can use PyTorch's save and load functions for the model parameters, and save the FLModel metadata with :mod:`FOBS<nvflare.fuel.utils.fobs>` separately:
```

Comment:
This is a bit confusing I think.

So to customize save & load.
Users can do either:
1. write a custom Persistor and then put that in the config file.
2. write a custom ModelController and overwrite the "save_model" "load_model" methods.

Do we really want to introduce all these different options?
I feel it should be better to keep it simple, so maybe provide option 1 is enough?

We should leave these "saving/loading/serialization" explanation to other section.

Reply:
Option 1 is good, however even a custom ModelPersistor is limiting as it is centered around ModelLearnable, and `save_model()`/`load_model()` don't have arguments for save location. I think Option 2 could be useful to showcase the flexibility?

Reply:
I see, but when you add additional argument to "save_model" and "load_model" that actually just overwrites them, they become new methods essentially.

Users can just add new methods call "save_model_with_location"?
Then in this case, maybe we don't even include these two methods in the class interface (model_controller.py)?
So remove "save_model" and "load_model" in "ModelController" class.

If we want to emphasis the saving / loading using persistor, maybe we need to move them up from BaseModelController to ModelController